libcaliptra/inc/caliptra_if.h

```diff
@@ -0,0 +1,54 @@
+//Licensed under the Apache-2.0 license
+
+#include <stdint.h>
+#include <stdbool.h>
+
+#define CALIPTRA_STATUS_OK 0
+
+// Memory
+
+/**
+ * caliptra_write_u32
+ *
+ * Writes a uint32_t value to the specified offset.
+ *
+ * @param[in] offset Memory offset to write
+ * @param[in] data   Data to write at offset
+ *
+ * @return 0 if successful, other if error (TBD)
+ */
+int caliptra_write_u32(uint32_t offset, uint32_t data);
+
+/**
+ * caliptra_read_u32
+ *
+ * Writes a uint32_t value to the specified offset.
+ *
+ * @param[in] offset Memory offset to read
+ * @param[in] data   Pointer to a uint32_t to store the data
+ *
+ * @return 0 if successful, other if error (TBD)
+ */
+int caliptra_read_u32(uint32_t offset, uint32_t *data);
+
+// Fuses
+
+/**
+ * caliptra_ready_for_fuses
+ *
+ * Reports if the Caliptra hardware is ready for fuse data
+ *
+ * @return bool True if ready, false otherwise
+ */
+bool caliptra_ready_for_fuses(void);
+
+// Firmware
```

Comment:
can we move those 2 functions into the caliptra_api.c and have a generic caliptra_wait() function which is implemented by the user.
Probably we can rename those as caliptra_wait_for_fuses(..), caliptra_wait_for_firmware(..)

Reply:
Right, and the underlying wait function would be defined within caliptra_if.h